cmd/kops/create_cluster.go

```diff
@@ -475,7 +475,7 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 
 		// Default to a DNS name for the bastion
 		cluster.Spec.Topology.Bastion = &api.BastionSpec{
-			BastionPublicName: "bastion-" + clusterName,
+			BastionPublicName: "bastion." + c.DNSZone,
```

Comment:
This is the reason the test is failing ...

```
--- FAIL: TestCreateClusterPrivate (0.03s)
	create_cluster_integration_test.go:164: diff:
		...
		      bastion:
		        enable: true
		+       name: bastion.
		-       name: bastion.private.example.com
		      masters: private
		      nodes: private
		...
		
	create_cluster_integration_test.go:166: YAML differed from expected
```

... which implies that `c.DNSZone` is empty. Which makes sense because I've tried to trace where this is set but not sure I understand what's going on:

1. We run `func TestCreateClusterPrivate` [here][1] ...
2. Which runs `func runCreateClusterIntegrationTest` [here][2] ...
3. Which initialises cluster parameters [here][3] but without `dns-zone` set, or a default ... 🤔 

The [DNS Zone docs seem to say][4] that there's some logic to determine it's value from cluster name if it's not set when creating the cluster ...

> If you don't specify a dns-zone, kops will list all your hosted zones, and choose the longest that
is a a suffix of your cluster name.  So for `dev.kubernetes.example.com`, if you have `kubernetes.example.com`,
`example.com` and `somethingelse.example.com`, it would choose `kubernetes.example.com`.  `example.com` matches but is shorter; `somethingelse.example.com` is not a suffix-match.

... but I can't seem to find where that's captured in the tests. 😰 Any ideas? 💡 

[1]: https://github.com/kubernetes/kops/blob/f7a7d4c9c41398c9c849731227f8f616e5ce0a7a/cmd/kops/create_cluster_integration_test.go#L49-L52
[2]: https://github.com/kubernetes/kops/blob/f7a7d4c9c41398c9c849731227f8f616e5ce0a7a/cmd/kops/create_cluster_integration_test.go#L54-L169
[3]: https://github.com/kubernetes/kops/blob/f7a7d4c9c41398c9c849731227f8f616e5ce0a7a/cmd/kops/create_cluster_integration_test.go#L84-L85
[4]: https://github.com/kubernetes/kops/blob/f7a7d4c9c41398c9c849731227f8f616e5ce0a7a/docs/arguments.md#dns-zone